### PR TITLE
Failing gradle build

### PR DESCRIPTION
**Pull request by l-1squared**

### CODE REVIEW

File: jgiven-gradle-plugin/src/main/java/com/tngtech/jgiven/gradle/internal/AbstractJGivenReportImpl.java [hunk 1]
```diff
@@ -106,5 +122,14 @@ public AbstractReportGenerator createGenerator() {
...
 
+    @Override
+    public Property<Boolean> getRequired() {
+        return isRequired;
+    }
+
+    @Override
+    public DirectoryProperty getOutputLocation() {
+        return null;
+    }
```
Comment (l-1squared) [inline jgiven-gradle-plugin/src/main/java/com/tngtech/jgiven/gradle/internal/AbstractJGivenReportImpl.java:133]: These are the methods that  the interface requires me to add

File: jgiven-gradle-plugin/src/main/java/com/tngtech/jgiven/gradle/internal/JGivenReportsContainerImpl.java [hunk 2]
```diff
@@ -11,7 +14,22 @@
...
+        System.out.println("DEBUG: created report");
+        String name = report.getName();
+        System.out.println("DEBUG: got report name");
+        if (name.equals("enabled")) {
+            throw new InvalidUserDataException("Reports that are part of a ReportContainer cannot be named 'enabled'");
+        } else {
+            System.out.println("DEBUG: evaluated name");
+            this.getStore().add(report);
+            this.index();
+        }
```
Comment (l-1squared) [inline jgiven-gradle-plugin/src/main/java/com/tngtech/jgiven/gradle/internal/JGivenReportsContainerImpl.java:30]: this code is a direct copy from the `add(JGivenHtmlReportImpl.class)` method. I could track the failure to line 20. I.e. where the new class is instantiated

File: jgiven-gradle-plugin/src/main/java/com/tngtech/jgiven/gradle/internal/AbstractJGivenReportImpl.java [hunk 3]
```diff
@@ -20,11 +26,21 @@
...
     private String title;
+    private final Property<Boolean> isRequired;
+    private final DirectoryProperty outputLocation;
     private boolean excludeEmptyScenarios = false;
     private boolean thumbnailsAreShown = true;
 
-    public AbstractJGivenReportImpl( String name, Task task, String relativeEntryPath ) {
+    public AbstractJGivenReportImpl(String name, Task task, String relativeEntryPath) {
         super( name, task, relativeEntryPath );
+        System.out.println("DEBUG 3:");
```
Comment (l-1squared) [inline jgiven-gradle-plugin/src/main/java/com/tngtech/jgiven/gradle/internal/AbstractJGivenReportImpl.java:36]: this statement never gets printed

File: jgiven-gradle-plugin/src/main/java/com/tngtech/jgiven/gradle/internal/AbstractJGivenReportImpl.java [hunk 4]
```diff
@@ -106,5 +122,14 @@ public AbstractReportGenerator createGenerator() {
...
     }
 
+    @Override
+    public Property<Boolean> getRequired() {
+        return isRequired;
+    }
+
+    @Override
+    public DirectoryProperty getOutputLocation() {
+        return null;
```
Comment (wolfs) [inline jgiven-gradle-plugin/src/main/java/com/tngtech/jgiven/gradle/internal/AbstractJGivenReportImpl.java:132]: Why does this return `null` and not `outputLocation`?

Comment (l-1squared) [reply jgiven-gradle-plugin/src/main/java/com/tngtech/jgiven/gradle/internal/AbstractJGivenReportImpl.java]: oops :blush: . Changed it, I am still getting an NPE

Comment (wolfs) [reply jgiven-gradle-plugin/src/main/java/com/tngtech/jgiven/gradle/internal/AbstractJGivenReportImpl.java]: @l-1squared Can you share a stacktrace?

Comment (l-1squared) [reply jgiven-gradle-plugin/src/main/java/com/tngtech/jgiven/gradle/internal/AbstractJGivenReportImpl.java]: @wolfs  I hope the trace I provided you via mail suffices. If not, I am happy to share a more complete log output tomorrow. However, the stacktrace I saw so far, did not help me very much.

**Comment by l-1squared** [comment]
Thanks @wolfs  for helping me out here. I have incorporated your suggestions into https://github.com/TNG/JGiven/pull/464 and will close this draft.

**Pull request closed without merging**